constraint/pkg/client/drivers/local/local.go

```diff
@@ -75,22 +86,104 @@ func New(args ...Arg) drivers.Driver {
 	for _, arg := range args {
 		arg(d)
 	}
+
+	// adding externaldata builtin otherwise capabilities get overridden
+	// if a capability, like http.send, is disabled
+	if d.providerCache != nil {
+		d.capabilities.Builtins = append(d.capabilities.Builtins, &ast.Builtin{
+			Name: "external_data",
+			Decl: opatypes.NewFunction(opatypes.Args(opatypes.A), opatypes.A),
+		})
+	}
 	d.compiler.WithCapabilities(d.capabilities)
 	return d
 }
 
 var _ drivers.Driver = &driver{}
 
 type driver struct {
-	modulesMux   sync.RWMutex
-	compiler     *ast.Compiler
-	modules      map[string]*ast.Module
-	storage      storage.Store
-	capabilities *ast.Capabilities
-	traceEnabled bool
+	modulesMux    sync.RWMutex
+	compiler      *ast.Compiler
+	modules       map[string]*ast.Module
+	storage       storage.Store
+	capabilities  *ast.Capabilities
+	traceEnabled  bool
+	providerCache *externaldata.ProviderCache
 }
 
 func (d *driver) Init(ctx context.Context) error {
+	if d.providerCache != nil {
+		rego.RegisterBuiltin1(
+			&rego.Function{
+				Name:    "external_data",
+				Decl:    opatypes.NewFunction(opatypes.Args(opatypes.A), opatypes.A),
+				Memoize: true,
+			},
+			func(bctx rego.BuiltinContext, regorequest *ast.Term) (*ast.Term, error) {
+				var regoReq externaldata.RegoRequest
+				if err := ast.As(regorequest.Value, &regoReq); err != nil {
+					return nil, err
+				}
+				// only primitive types are allowed for keys
+				for _, key := range regoReq.Keys {
+					switch v := key.(type) {
+					case int:
```

Comment:
do we need to worry about `int32`, `int64`?

Reply:
added `int32` and `int64`. Omitted `bool` here, not sure if there would be a usecase for that.

Reply:
ack